__tests__/src/diff-snapshot.spec.js

```diff
@@ -80,53 +62,74 @@ describe('diff-snapshot', () => {
             return !!defaultExists;
         }
       });
+      mockFs.readFileSync.mockImplementation((p) => {
+        const bn = path.basename(p);
+
+        if (bn === 'id1-snap.png' && snapshotExists) {
+          return mockImageBuffer;
+        }
+
+        return null;
+      });
+
+      jest.mock('pixelmatch', () => mockPixelMatch);
+      mockPixelMatch.mockImplementation(() => pixelmatchResult);
 
       return diffImageToSnapshot;
     }
 
-    it('should throw if an unsupported configuration is passed', () => {
-      const diffImageToSnapshot = setupTest({});
-      expect(() =>
-        diffImageToSnapshot({
-          imageData: mockImageBuffer,
-          snapshotIdentifier: mockSnapshotIdentifier,
-          snapshotsDir: mockSnapshotsDir,
-          customDiffConfig: {
-            imageOutputPath: 'path/to/output/dir',
-          },
-        })
-      ).toThrowErrorMatchingSnapshot();
-    });
-
     it('should run comparison if there is already a snapshot stored and updateSnapshot flag is not set', () => {
       const diffImageToSnapshot = setupTest({ snapshotExists: true });
-      diffImageToSnapshot({
+      const result = diffImageToSnapshot({
         imageData: mockImageBuffer,
         snapshotIdentifier: mockSnapshotIdentifier,
         snapshotsDir: mockSnapshotsDir,
         updateSnapshot: false,
       });
 
-      expect(mockRunSync).toHaveBeenCalled();
+      expect(result).toMatchSnapshot();
+      expect(mockPixelMatch).toHaveBeenCalled();
+      expect(mockWriteFileSync).not.toHaveBeenCalled();
+    });
+
+    it('should write a diff image only if the test fails', () => {
+      const diffImageToSnapshot = setupTest({ snapshotExists: true, pixelmatchResult: 5000 });
+      const result = diffImageToSnapshot({
+        imageData: mockFailImageBuffer,
+        snapshotIdentifier: mockSnapshotIdentifier,
+        snapshotsDir: mockSnapshotsDir,
+        updateSnapshot: false,
+      });
+
+      expect(result).toMatchSnapshot();
+      expect(mockPixelMatch).toHaveBeenCalled();
+      expect(mockWriteFileSync).toHaveBeenCalled();
```

Comment:
Be more specific in validating what this got called with.

Reply:
I've learnt `expect.any(Buffer)` is my friend